feat/Sales page

Salvador511 opened the pull request:
This pull request introduces several enhancements and new features to improve the purchasing and sales modules, including table components, modal dialogs, and backend API integration. Key changes include adding tables for displaying lots and suppliers, implementing modal dialogs for editing and adding new items, and modifying the backend API to include product details in lot queries.

### Backend Enhancements:
* Updated `src/app/api/lot/route.js` to include product details in the `findMany` query for lots, enabling richer data display in the frontend.

### Purchasing Module Improvements:
* Added `LotsTable` and `SuppliersTable` components in `src/app/purchasing/LotsTable.jsx` and `src/app/purchasing/SuppliersTable.jsx`, respectively, to display paginated tables for lots and suppliers with search functionality. [[1]](diffhunk://#diff-57de6656a30a03f9ab9308a60859e411c7f3c8053e9ce25e06889fbf28fecd95R1-R87) [[2]](diffhunk://#diff-78faf3a244b0194b1d2f681cd4b8386655f24655a0832fd745dc628217144a26R1-R85)
* Created `ModalLots` component in `src/app/purchasing/ModalLots.jsx` for adding new products via a modal dialog with styled inputs and validation placeholders.
* Integrated the purchasing module's tables and modal dialogs into a unified page in `src/app/purchasing/page.jsx`, with tabs for switching between lots and suppliers.

### Sales Module Enhancements:
* Added `MGoodReceipt` and `MSuppliers` components in `src/app/sales/components/MGoodReceipt.jsx` and `src/app/sales/components/MSuppliers.jsx`, respectively, for creating new goods receipts and suppliers via modal dialogs. [[1]](diffhunk://#diff-4cb1e94fe2937c6b27906deb3b981f24bbb014af558d3ecc7a91ae0f46efd796R1-R57) [[2]](diffhunk://#diff-bffc5f2c2fea0b1de4890c5b528afdf462cfb93d05da4b7c16cf1506e371f5a7R1-R28)
* Created a reusable `Table` component in `src/app/sales/components/table.jsx` for displaying goods receipt and supplier data, with dynamic column rendering based on the table type.
* Implemented a generic `Modal` component in `src/app/sales/components/modal.jsx` for consistent modal styling and behavior across the sales module.

### Development Configuration:
* Added `.vscode/c_cpp_properties.json` to configure C/C++ IntelliSense settings for the workspace, improving development efficiency.

FerIbarra04 committed:
create Sales and components (files)

FerIbarra04 committed:
Plantilla Front

Se hizo modal, tabs se agregaron colores, faltan detalles de colores acomodo y trans

Commit:
avance de tablas y una modal

Salvador511 committed:
feat: implement add lot and add supplier modals with validation

Salvador511 committed:
fix: remove unused components and modal files

DonAnchondoBis reviewed (changes requested):
Necesitamos terminar la otra modal

Salvador511 committed:
feat: add phone number validation for supplier schema

Salvador511 committed:
fix: update purchasing links and enhance data fetching in modals

FerIbarra04 committed:
ModalAddSupplier, page.jsx and AuthWrapper

Se termino la modal de supp y su fetch, se movio la linea 262 de page y el AuthWrapper

FerIbarra04 committed:
AuthWrapper

.

FerIbarra04 committed:
AW, pa, rb

.

FerIbarra04 committed:
Spaces...

...

Salvador511 committed:
fix: update valid roles for supplier access to include WAREHOUSE

Salvador511 committed:
fix: refactor ModalAddSupplier component for improved form structure and error handling

Salvador511 committed:
fix: update fetchData call in Wrapper component to remove entity parameter

DonAnchondoBis commented:
<h2>Coverage Report</h2>
<table> <thead> <tr> <th align="center">Status</th> <th align="left">Category</th> <th align="right">Percentage</th> <th align="right">Covered / Total</th> </tr> </thead> <tbody> <tr> <td align="center">🔵</td> <td align="left">Lines</td> <td align="right">34.83%<br/>⬇️ <em>-13.42%</em></td> <td align="right">814 / 2337</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Statements</td> <td align="right">34.83%<br/>⬇️ <em>-13.42%</em></td> <td align="right">814 / 2337</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Functions</td> <td align="right">80%<br/>⬇️ <em>-3.05%</em></td> <td align="right">52 / 65</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Branches</td> <td align="right">89.91%<br/>⬇️ <em>-0.70%</em></td> <td align="right">312 / 347</td> </tr> </tbody> </table><details><summary>File Coverage</summary><table> <thead> <tr> <th align="left">File</th> <th align="right">Stmts</th> <th align="right">Branches</th> <th align="right">Functions</th> <th align="right">Lines</th> <th align="left">Uncovered Lines</th> </tr> </thead> <tbody> <tr> <td align="left" colspan="6"><b>Changed Files</b></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/api/supplier/route.js">src/app/api/supplier/route.js</a></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="left"></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/landingPageComponents/Dashboard/Dashboard.jsx">src/app/landingPageComponents/Dashboard/Dashboard.jsx</a></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">100%<br/>🟰 <em>±0%</em></td> <td align="right">0%<br/>🟰 <em>±0%</em></td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/landingPageComponents/Dashboard/Dashboard.jsx#L3-L128" class="text-red">3-128</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/LotsTable.jsx">src/app/purchasing/LotsTable.jsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/LotsTable.jsx#L1-L77" class="text-red">1-77</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/ModalAddLot.jsx">src/app/purchasing/ModalAddLot.jsx</a></td> <td align="right">0%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/ModalAddLot.jsx#L2-L152" class="text-red">2-152</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/ModalAddSupplier.jsx">src/app/purchasing/ModalAddSupplier.jsx</a></td> <td align="right">0%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/ModalAddSupplier.jsx#L2-L152" class="text-red">2-152</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/SuppliersTable.jsx">src/app/purchasing/SuppliersTable.jsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/SuppliersTable.jsx#L1-L77" class="text-red">1-77</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/page.jsx">src/app/purchasing/page.jsx</a></td> <td align="right">0%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/page.jsx#L2-L286" class="text-red">2-286</a></td> </tr> <tr> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/utils.js">src/app/purchasing/utils.js</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/blob/9ef0023ea81650d9ff90d5f1f78d6409c676c703/src/app/purchasing/utils.js#L1-L32" class="text-red">1-32</a></td> </tr> </tbody> </table></details>
<em>Generated in workflow <a href=https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/actions/runs/15762752979>#262</a> for commit <a href="https://github.com/DonAnchondoBis/AbarrotesDonAnchondo/commit/9ef0023ea81650d9ff90d5f1f78d6409c676c703">9ef0023</a> by the <a href="https://github.com/davelosert/vitest-coverage-report-action">Vitest Coverage Report Action</a></em>

<!-- vitest-coverage-report-marker-root -->
